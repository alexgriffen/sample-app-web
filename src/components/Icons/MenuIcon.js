import React, { Component } from 'react';
import { IconButton, withStyles } from '@material-ui/core';
import * as PropTypes from 'prop-types';
import menuIconPng from '../../assets/img/menu.png';
import menuIconSvg from '../../assets/svg/menu@3x.svg';
import { colors } from '../../utils/Colors';

const styles = () => ({
    menuIcon: {
      background: `url(${ menuIconPng }) no-repeat center center`,
      backgroundImage: `url(${menuIconSvg})`,
      borderRadius: 0,
      height: 40,
      width: 40,
      '&:hover': {
        backgroundColor: colors.white,
      },
    },
  }
);
const { func, object, string } = PropTypes;

class MenuIcon extends Component {
  static propTypes = {
    classes: object.isRequired,
    fallBackClasses: string,
    testID: string,
    onClick: func.isRequired,
  };

  static defaultProps = {
    fallBackClasses: '',
    testID: 'menu-icon-id',
  };

  render() {
    const { classes, fallBackClasses, onClick, testID } = this.props;

    return (
      <IconButton
        classes={ {
          // ` ${fallBackClasses}` is there to make it backwards compatible,
          // it doesn't not have an other function
          root: classes.menuIcon + ` ${ fallBackClasses }`,
        } }
        id={ testID }
        onClick={ onClick }
      />
    );
  }
}

export default withStyles(styles)(MenuIcon);
