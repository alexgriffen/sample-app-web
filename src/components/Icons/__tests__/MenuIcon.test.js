import React from 'react';
import { shallow } from 'enzyme/build';
import MenuIcon from '../MenuIcon';

let props;

describe('MenuIcon', () => {
  beforeEach(() => {
    props = {
      onClick: jest.fn(),
    };
  });

  it('should render correctly with the required options', () => {
    const component = shallow(<MenuIcon { ...props } />).dive();

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with full options', () => {
    const component = shallow(
      <MenuIcon
        { ...props }
        fallBackClasses="fallback-classes"
        testID="foo-test-bar-id"
      />
    ).dive();

    expect(component).toMatchSnapshot();
  });
});
