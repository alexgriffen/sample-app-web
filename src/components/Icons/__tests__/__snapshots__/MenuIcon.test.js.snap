// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuIcon should render correctly with full options 1`] = `
<WithStyles(IconButton)
  classes={
    Object {
      "root": "MenuIcon-menuIcon-1 fallback-classes",
    }
  }
  id="foo-test-bar-id"
  onClick={[MockFunction]}
/>
`;

exports[`MenuIcon should render correctly with the required options 1`] = `
<WithStyles(IconButton)
  classes={
    Object {
      "root": "MenuIcon-menuIcon-1 ",
    }
  }
  id="menu-icon-id"
  onClick={[MockFunction]}
/>
`;
